precompiles/xcm/src/lib.rs

```diff
@@ -479,30 +522,47 @@ where
 
         // Read call arguments
         let currency_address = input.read::<Address>()?;
-        let amount_of_tokens = input
+        let amount_of_tokens: u128 = input
             .read::<U256>()?
             .try_into()
             .map_err(|_| revert("error converting amount_of_tokens, maybe value too large"))?;
         let destination = input.read::<MultiLocation>()?;
         let weight = input.read::<WeightV2>()?;
 
-        let asset_id = Runtime::address_to_asset_id(currency_address.into())
-            .ok_or(revert("Failed to resolve fee asset id from address"))?;
         let dest_weight_limit = if weight.is_zero() {
             WeightLimit::Unlimited
         } else {
             WeightLimit::Limited(weight.get_weight())
         };
 
-        log::trace!(target: "xcm-precompile::transfer", "Raw arguments: currency_address: {:?}, amount_of_tokens: {:?}, destination: {:?}, \
-        weight: {:?}, calculated asset_id: {:?}",
-        currency_address, amount_of_tokens, destination, weight, asset_id);
+        let call = {
+            if currency_address == Address::from(NATIVE_ADDRESS) {
```

Comment:
What about `transfer_with_fee` call?

Reply:
You are right, we should change that as well. In earlier discussion, we only talked about changing transfer but this should be consistent. I will change it.